### ISSUE TITLE
Support node api documents?

### ISSUE THREAD

**Issue by southorange0929** [web-infra-dev/rspack#3047]
### What problem does this feature solve?

Just like webpack, it provide node api documents for us to custom pack logic.

### What does the proposed API of configuration look like?

https://webpack.js.org/api/node

**Comment by hyf0** [comment]
Definitely, something we want to do, but we currently lack manpower :(.
